packages/pkl.experimental.structuredRead/tests/dynamicType.pkl

```diff
@@ -0,0 +1,70 @@
+//===----------------------------------------------------------------------===//
+// Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+module pkl.experimental.structuredRead.tests.dynamicType
+amends "pkl:test"
+
+import ".../structuredRead.pkl"
+
+local function testHarness(name: String, value: String) = (structuredRead.fromExternalProps) {
+  local responses = Map(name, value)
+  readFunc = (s:String) -> responses.getOrNull(s.replaceFirst(inputScheme, ""))
```

Comment:
Although this is a legit workaround and it will remain necessary for `fromHttps`, this is a sizeable part of the reason `pkl:Project` [offers `externalProperties` and `env`](https://pkl-lang.org/package-docs/pkl/current/EvaluatorSettings/index.html#externalProperties).

Reply:
I took a look at that module, but it's clear to me how to use it in conjunction with `pkl test`. Do we have any examples of these two working together?

Reply:
`PklProject`:
```
amends "pkl:Project"

evaluatorSettings {
  env {
    ["FOO"] = "bar"
  }
}

tests {
  ...import*("*test.pkl").keys
}
```

`my_test.pkl`:
```
amends "pkl:test"

facts {
  ["Can read env from project"] {
    read("env:FOO") == "bar"
  }
}
```

and now you can either just run `pkl test` (in the project dir, or with a `--project-dir`), or you can `pkl test my_test.pkl` (again, in a way that lets Pkl find the project).